### PR TITLE
adding the test train split as well as the module 2.train-VAE. note t…

### PR DESCRIPTION
**Pull request by mjw-18**
…hat module 2.train-VAE is incomplete and is still in progress. this will be updated as we progress towards the training phase. there was also a change made to the data_loader.py file so that the output data frame will now be reindexed.

### CODE REVIEW

**Comment by mjw-18** [comment]
@gwaybio check out the changes! these changes are certainly up for review and the feedback will be very valuable

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: Please use [`train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) from scikit-learn.

It is not good practice to take only the top x number of samples (where x represents a certain percentage). 1) We do not know how the samples are ordered in that dataframe, and 2) hardcoding the number of samples (rather than a percentage) does not buffer against scenarios where the number of samples changes.

See [here](https://github.com/greenelab/pancancer/blob/d1b3de7fa387d0a44d0a4468b0ac30918ed66886/scripts/pancancer_classifier.py#L275-L278) for an example

Comment (gwaybio) [reply 1.data-exploration/scripts/nbconverted/test-train_split.py]: this small command will replace most of the code here and is much more sustainable - i highly recommend it!

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: Please use only minimal package imports - only import code you need. (e.g. PCA isn't used, so please remove import)

File: 2.train-VAE/utils.py
Comment (gwaybio) [inline 2.train-VAE/utils.py]: I don't think you need this file. Please make sure to review the files that you're adding

File: 2.train-VAE/vae.py
Comment (gwaybio) [inline 2.train-VAE/vae.py:3]: Add a citation for where you copied this code from

Comment (gwaybio) [reply 2.train-VAE/vae.py:3]: (and a link)

File: 2.train-VAE/vae_utils.py
Comment (gwaybio) [inline 2.train-VAE/vae_utils.py:3]: Add a citation to where you copied this code from (see comment above)

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py:106]: you need to somehow save this info - probably in a new column in the sample_info file

**Review by gwaybio — CHANGES_REQUESTED** [review]
A couple comments that require addressing.

Also, please do not merge this code prior to receiving an "approval" - you merged #6 prematurely.

File: 2.train-VAE/vae.py
Comment (mjw-18) [reply 2.train-VAE/vae.py:3]: this import will actually be removed as it wasnt utilized in the code!

File: 2.train-VAE/vae_utils.py
Comment (mjw-18) [reply 2.train-VAE/vae_utils.py:3]: this will also be removed for same reason as above

File: 2.train-VAE/vae.py
Comment (mjw-18) [reply 2.train-VAE/vae.py:3]: my bad greg I noticed that you were referring to the .py file itself rather than import. I'll add all proper citation to the appropriate .py files

File: 2.train-VAE/utils.py
Comment (mjw-18) [reply 2.train-VAE/utils.py]: got it! I made the mistake of copying all the .py files into the module doing the trial and error method when trying to fix the beta VAE notebook.

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py:41]: ah! One thing to confirm here: Right now you don't have a definitive check that the dependency rows are exactly aligned with the sample info file.

How can you confirm that they are the same order?

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: I'm not sure if you need to do this.

Think about what you actually need for train test split. Why do you need this logic?

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: I think this comment is out of date/incorrect

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: I think this comment is out of date/incorrect

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py:83]: This is PERFECT 💯 

Think about what you're doing here, and decide if what you're doing to get PA_dependency_df ready is necessary.

I think that you're doing a bit too much wrangling, and you need to confirm sample ID alignment.

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: rather than printing the whole column, It's best to do two things:

1. Print the shape: `print(test_df.shape)`
2. Print only the top couple columns: `test_df.head(3)`

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: I think you probably want to save with `index=False`: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: same comments I mentioned for test_df

File: 2.train-VAE/vae.py
Comment (gwaybio) [inline 2.train-VAE/vae.py]:
```suggestion
# THIS CODE WAS COPIED FROM THE FOLLOWING URL: https://github.com/broadinstitute/cell-painting-vae/blob/master/scripts/vae.py
```

File: 2.train-VAE/vae_utils.py
Comment (gwaybio) [inline 2.train-VAE/vae_utils.py]:
```suggestion
    return 

```

File: 2.train-VAE/vae.py
Comment (gwaybio) [inline 2.train-VAE/vae.py]:
```suggestion
        )

```

.py files need an extra new line at the EOF

File: 2.train-VAE/scripts/nbconverted/beta_vae_training.py
Comment (gwaybio) [inline 2.train-VAE/scripts/nbconverted/beta_vae_training.py:64]: you're currently training with the full data, you probably only want to train with the train data :)

we're also a couple steps ahead of ourselves with adding the actual training (subset genes, hyperparameter search)

**Review by gwaybio — CHANGES_REQUESTED** [review]
Sooo close. Very well done and lots of improvements. Let's not stop now :)

You should think through the train_test_split one last time. I made very specific comments that would be worth thinking deeply through.

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [reply 1.data-exploration/scripts/nbconverted/test-train_split.py]: (sorry, this comment is supposed to refer to lines 53-70)

File: 0.data-download/scripts/data_loader.py
Comment (gwaybio) [inline 0.data-download/scripts/data_loader.py]: This logic looks good, but just noting that all of your code above will execute (loading, subsetting data) with no effect if you enter these if statements.

Also, you probably should create a new variable (setting `adult_or_pediatric` ="test" or "train" doesn't really make sense)

Comment (gwaybio) [reply 0.data-download/scripts/data_loader.py]: the solution might be to create an entirely different data load function (`load_train_test_data()`) or something like that

```python
train_df, test_df = load_train_test_data(data_dir=PATH)
```

File: 1.data-exploration/scripts/nbconverted/test-train_split.py
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/test-train_split.py]: this line is not having an effect, btw (you're not setting it to a variable)

**Review by gwaybio — APPROVED** [review]
Please build a new function for loading train and test data - see comment below.

Once you add, feel free to merge

**Pull request merged**
